### ISSUE TITLE
`update_fee` not acked when shutting down

### ISSUE THREAD

**Issue by t-bast** [ElementsProject/lightning#4714]
I've been investigating a mutual close that went wrong between the ACINQ node and a c-lightning node.
The gist of the issue is that when an `update_fee` has been signed before a `shutdown`, c-lightning doesn't send the final `revoke_and_ack` so we disagree on when we should move to the negotiating phase (exchanging `closing_signed`).

The flow of messages exchanged was:

```
  c-lightning                eclair
       |                        |
       |      update_fee        |
       |----------------------->|
       |      commit_sig        |
       |----------------------->|
       |       shutdown         |
       |----------------------->|
       |    revoke_and_ack      |
       |<-----------------------|
       |       shutdown         |
       |<-----------------------|
       |      commit_sig        |
       |<-----------------------|
       |     closing_signed     |
       |----------------------->|
```

At that point, eclair waits for the `revoke_and_ack` before going to the `negotiating` step and sending its own `closing_signed`, but c-lightning never sends it. Since there aren't any pending htlcs, this shouldn't be strictly necessary, but our state machine insists that updates that were signed before initiating shutdown must complete before signing closing txs.

Do you think c-lightning could send its `revoke_and_ack` before sending `closing_signed` in that case?
Or should eclair be more lax and move to the negotiating phase even if a pending `update_fee` is not complete?

PS: damn I can't wait to completely remove `update_fee`, life will be so much simpler when feerate is always zero and we set it at closing time (with CPFP or RBF)!

**Comment by rustyrussell** [comment]
AI cleanup of old issues.  If wrong, please open a new issue.


Triage decision summary:
Issue about update_fee message handling during shutdown in mutual close. The issue reporter notes in PS that they expect update_fee to be completely removed in future versions. Given the age (2021) and the stated intent to remove update_fee functionality, this is likely obsolete or superseded by protocol changes.
